src/addDefaultProps.coffee

```diff
@@ -0,0 +1,21 @@
+import {isFunction} from './util/helpers'
+import addProps from './addProps'
+
+addDefaultProps = (createDefaults) ->
+  addProps (props) ->
+    defaults = if isFunction createDefaults
+      createDefaults props
+    else
+      createDefaults
+
+    Object.keys(defaults).reduce(
+      (newProps, key) ->
+        if props[key] is null or props[key] is undefined
```

Comment:
In JS, I'd write this test (for "nullishness") as `== null`

But in Coffeescript you can use the `?` operator, see above comment

However I think it's worth considering allowing them to inject their own custom "is missing" callback (in case eg they'd rather have only `undefined` trigger default props), in which case I'd vote for a signature like eg:
```
addDefaultProps(
  {
    a: 1,
  },
  {
    isMissingFunc: val => val === undefined,
  }
)
```

Reply:
I've made all the suggested changes other the the custom "is missing" callback one. Do you imagine that being something that's called for each default prop? eg.

```js
addDefaultProps({ a: 1, b: 2}, { isMissing: (key, value) => ... })
```

… or just once for all the default props? Neither way seems completely obvious to me.

Reply:
Ya I pictured it being a callback invoked for each prop separately

So eg:
```
addDefaultProps({a: 1, b: 2}, {isMissing: val => val === undefined})({a: undefined, b: null}) // {a: 1, b: null}
```

Your example makes a good point that `key` should also be supplied in case you want it, though I'd go with the Lodash-style `(val, key) ->` signature

That being said, I'm fine with not including this option until someone needs it (it'd be a nonbreaking API change)

Reply:
Personally, that seems complicated/non-obvious enough that I’d probably just do it the “normal” way with `addProps` instead, rather than dealing with a different API.